src/main/java/pl/edu/mimuw/Formula.java

```diff
@@ -0,0 +1,10 @@
+package pl.edu.mimuw;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+public abstract class Formula {
+
+  public abstract boolean eval(HashMap<String, Boolean> data);
```

Comment:
nawej jak to się stało klasą abstrakcyjną, lepiej by miało jako parametr metody `Map` bo jest ogólniejsze